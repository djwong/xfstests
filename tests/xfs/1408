#! /bin/bash
# FS QA Test No. 1408
#
# Populate a XFS filesystem and fuzz every leafn-format dir block field.
# Do not fix the filesystem, to test metadata verifiers.
#
#-----------------------------------------------------------------------
# Copyright (c) 2018 Oracle, Inc.  All Rights Reserved.
#
# This program is free software; you can redistribute it and/or
# modify it under the terms of the GNU General Public License as
# published by the Free Software Foundation.
#
# This program is distributed in the hope that it would be useful,
# but WITHOUT ANY WARRANTY; without even the implied warranty of
# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
# GNU General Public License for more details.
#
# You should have received a copy of the GNU General Public License
# along with this program; if not, write the Free Software Foundation,
# Inc.,  51 Franklin St, Fifth Floor, Boston, MA  02110-1303  USA
#-----------------------------------------------------------------------
#

seq=`basename $0`
seqres=$RESULT_DIR/$seq
echo "QA output created by $seq"

here=`pwd`
tmp=/tmp/$$
status=1	# failure is the default!
trap "_cleanup; exit \$status" 0 1 2 3 7 15

_cleanup()
{
	cd /
	rm -rf $tmp.*
}

# get standard environment, filters and checks
. ./common/rc
. ./common/filter
. ./common/populate
. ./common/fuzzy

# real QA test starts here
_supported_os Linux
_supported_fs xfs
_require_scratch_xfs_fuzz_fields

echo "Format and populate"
_scratch_populate_cached nofill > $seqres.full 2>&1

echo "Find leafn-format dir block"
_scratch_mount
inum=$(stat -c '%i' $SCRATCH_MNT/S_IFDIR.FMT_NODE)
blk_sz=$(_get_block_size $SCRATCH_MNT)
_scratch_unmount

leaf_offset=$(( ( (2 ** 35) / blk_sz) + 1))
echo "Fuzz leafn-format dir block"
_scratch_xfs_fuzz_metadata '' 'none'  "inode ${inum}" "dblock ${leaf_offset}" >> $seqres.full
echo "Done fuzzing leafn-format dir block"

# success, all done
status=0
exit
