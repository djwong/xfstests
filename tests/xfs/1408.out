QA output created by 1408
Format and populate
Find leafn-format dir block
Fuzz leafn-format dir block
Done fuzzing leafn-format dir block
